cmake/stm32/common.cmake

```diff
@@ -10,11 +10,17 @@ foreach(FAMILY ${STM32_SUPPORTED_FAMILIES_LONG_NAME})
 endforeach()
 list(REMOVE_DUPLICATES STM32_SUPPORTED_FAMILIES_SHORT_NAME)
 
-if(NOT STM32_TOOLCHAIN_PATH)
-     set(STM32_TOOLCHAIN_PATH "/usr")
-     message(STATUS "No STM32_TOOLCHAIN_PATH specified, using default: " ${STM32_TOOLCHAIN_PATH})
+if(NOT STM32_TOOLCHAIN_PATH AND NOT CMAKE_C_COMPILER)
+    set(STM32_TOOLCHAIN_PATH "/usr")
+    message(STATUS "No STM32_TOOLCHAIN_PATH specified, using default: " ${STM32_TOOLCHAIN_PATH})
 else()
-     file(TO_CMAKE_PATH "${STM32_TOOLCHAIN_PATH}" STM32_TOOLCHAIN_PATH)
+    if(NOT STM32_TOOLCHAIN_PATH)
+        # keep only directory of compiler
+        get_filename_component(STM32_TOOLCHAIN_PATH ${CMAKE_C_COMPILER} DIRECTORY)
+        # remove the last /bin directory
+        get_filename_component(STM32_TOOLCHAIN_PATH ${STM32_TOOLCHAIN_PATH} DIRECTORY)
+    endif()
+    file(TO_CMAKE_PATH "${STM32_TOOLCHAIN_PATH}" STM32_TOOLCHAIN_PATH)

```

Comment:
The logic is confusing
```
if(not a && not b)
else() # if(a or b)
    if(not a)
    endif()
endif()
```

Reply:
I did my best. What do you propose ?

Reply:
What about :
```
if(not a && not b)
elseif(not a) # the `&& b` might be a good to  
endif()
file(TO_CMAKE_PATH "${STM32_TOOLCHAIN_PATH}" STM32_TOOLCHAIN_PATH)
```

I think doing the conversion `TO_CMAKE_PATH` is good either way, so I took it out of the if blocks.
What do you think?

Reply:
What's good after 3 month is that I'm stranger to my code and can judge it better. Sorry for delay :)